test/units/testsuite-64.sh

```diff
@@ -658,24 +625,16 @@ testcase_iscsi_lvm() {
     # "Reset" the DM state, since we yanked the backing storage from under the LVM,
     # so the currently active VGs/LVs are invalid
     dmsetup remove_all --deferred
-    udevadm settle
     # The LVM and iSCSI related symlinks should be gone
-    test ! -e "/dev/$vgroup"
-    test ! -e "/dev/disk/by-label/mylvpart1"
-    for link in "${expected_symlinks[@]}"; do
-        test ! -e "$link"
-    done
+    udevadm wait --settle --timeout=30 --removed "/dev/$vgroup" "/dev/disk/by-label/mylvpart1" "${expected_symlinks[@]}"
     helper_check_device_symlinks "/dev/disk"
     # Reconnect the iSCSI devices and check if everything get detected correctly
     iscsiadm --mode discoverydb --type sendtargets --portal "$target_ip" --discover
     iscsiadm --mode node --targetname "$target_name" --portal "$target_ip:$target_port" --login
-    udevadm settle
+    udevadm wait --settle --timeout=30 "${expected_symlinks[@]}"
```

Comment:
do we really need `--settle` on these?

Reply:
Yes. Not only 'add' but 'change' event may be already queued for the devices.